buffer/src/main/java/io/netty/buffer/SizeClasses.java

```diff
@@ -329,7 +329,7 @@ public int size2SizeIdx(int size) {
             return size2idxTab[size - 1 >> LOG2_QUANTUM];
         }
 
-        int x = log2((size << 1) - 1);
+        int x = 32 - Integer.numberOfLeadingZeros(size - 1);
```

Comment:
But, the orignal code, whcih was using 
https://github.com/netty/netty/blob/f1fa227ddf675f055766d04900cce6804fd7f710/buffer/src/main/java/io/netty/buffer/PoolThreadCache.java#L148

inlined, would become:

```java
int x = (31 - Integer.numberOfLeadingZeros(size << 1) )  -1;
```
while, with this change:
```java
int x = 32 - Integer.numberOfLeadingZeros(size - 1);
```
which is saving just a shift op which is already super cheap, really and is not reusing anymore the existing `log2` method.
This means that it will inlined for sure (regardless inlining decisions of the JIT), but still make the code a bit less clear to me.